napari/view_layers.py

```diff
@@ -1,4 +1,11 @@
 from .viewer import Viewer
+from .utils.magic import magic_name
+
+
+def _guess_name(name, data):
```

Comment:
if `_guess_name` is getting called in the `AddLayersMixin` why do we need a copy here?

Reply:
yes, as it will guess based on the calling function, which means that if we didn’t do this in `view_*`, it would always default to `'data'`

Reply:
ah, hmm - it doesn't seem so desirable to have the duplicate code. Maybe at least we can move `guess_name` to our utils too?

Reply:
yeah I was thinking that too. also note to self: need to fix the assignment tests